packages/client/src/grant-token.ts

```diff
@@ -0,0 +1,32 @@
+import axios from 'axios';
+import { z } from 'zod';
+
+const TokenSetParametersSchema = z.object({
+  access_token: z.string(),
+  expires_in: z.number(),
+  id_token: z.string(),
+  refresh_token: z.string(),
+});
+
+export type TokenSetParameters = z.infer<typeof TokenSetParametersSchema>;
+
+export const grantTokenByAuthorizationCode = async (
+  endpoint: string,
+  code: string,
+  redirectUri: string,
+  codeVerifier: string
+): Promise<TokenSetParameters> => {
+  const parameters = new URLSearchParams();
+  parameters.append('grant_type', 'authorization_code');
+  parameters.append('code', code);
+  parameters.append('redirect_uri', redirectUri);
+  parameters.append('code_verifier', codeVerifier);
+
+  const { data } = await axios.post<TokenSetParameters>(endpoint, parameters, {
```

Comment:
TODO: error handle?

Reply:
feels like we should have a general error handling solution for all client api calls